corehq/apps/app_manager/models.py

```diff
@@ -3399,7 +3399,20 @@ def _column_to_series(column):
                             xpath=suite_xml.Xpath(function='description'))
                     ),
                 ),
-            ] + list(_get_graph_fields())
+            ] + list(_get_graph_fields()) + [
+                suite_xml.Field(
+                    header=suite_xml.Header(
+                        text=suite_xml.Text(
+                            locale=suite_xml.Locale(id=id_strings.report_last_sync())
+                        )
+                    ),
+                    template=suite_xml.Template(
+                        text=suite_xml.Text(
+                            xpath=suite_xml.Xpath(function="instance('reports')/reports/@last_sync")
```

Comment:
Sorry, to confirm: This is for UCR and mobile reporting?

As long as you include the timezone info in the ISO date, it won't matter
what the specific timezone is (we normalize to UTC and strip).

The format-date method should coerce a date to the local timezone on
display, so I think we're good doing the best practice here as long as we
use that function.

-Clayton

On Tue, Sep 8, 2015 at 5:25 PM, Daniel Roberts <notifications@github.com>
wrote:

> In corehq/apps/app_manager/models.py
> <https://github.com/dimagi/commcare-hq/pull/8132#discussion_r38982313>:
>
> > @@ -3399,7 +3399,20 @@ def _column_to_series(column):
> >                              xpath=suite_xml.Xpath(function='description'))
> >                      ),
> >                  ),
> > -            ] + list(_get_graph_fields())
> > +            ] + list(_get_graph_fields()) + [
> > +                suite_xml.Field(
> > +                    header=suite_xml.Header(
> > +                        text=suite_xml.Text(
> > +                            locale=suite_xml.Locale(id=id_strings.report_last_sync())
> > +                        )
> > +                    ),
> > +                    template=suite_xml.Template(
> > +                        text=suite_xml.Text(
> > +                            xpath=suite_xml.Xpath(function="instance('reports')/reports/@last_sync")
>
> @ctsims <https://github.com/ctsims>—is it better for HQ to send down a
> datetime formatted for the user (shifted to their timezone, etc.), the way
> it's doing in this PR, or is it better to send down an ISO-8601 UTC string
> and let the phone deal with the shifting? The second sounds better to me,
> but I want to confirm that the detail screen will recognize it as a date
> (or we can coerce it to one) and that'll display it in the user's perferred
> TZ and format?
>
> This is the template/text/xpath function proposed (@last_sync is a
> formatted datetime string to be shown directly to the user). If @last_sync
> were instead a ISO-8601 UTC string, what (if anything) should this be to
> make that appear correctly?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dimagi/commcare-hq/pull/8132/files#r38982313>.
>